pkgs/top-level/impure.nix

```diff
@@ -40,18 +40,25 @@ in
   # collections of packages.  These collection of packages are part of the
   # fix-point made by Nixpkgs.
   overlays ? let
-      dirPath = try (if pathExists <nixpkgs-overlays> then <nixpkgs-overlays> else "") "";
-      dirHome = homeDir + "/.config/nixpkgs/overlays";
-      dirCheck = dir: dir != "" && pathExists (dir + "/.");
-      overlays = dir:
-        let content = readDir dir; in
-        map (n: import (dir + ("/" + n)))
-          (builtins.filter (n: builtins.match ".*\.nix" n != null || pathExists (dir + ("/" + n + "/default.nix")))
-            (attrNames content));
+      pathOverlays = try (if pathExists <nixpkgs-overlays> then <nixpkgs-overlays> else "") "";
+      homeOverlaysDir = homeDir + "/.config/nixpkgs/overlays";
+      homeOverlaysFile = homeDir + "/.config/nixpkgs/overlays.nix";
+      overlays = path:
+        # check if the path is a directory or a file
+        if pathExists (path + "/.")
+        # it's a directory, so the set of overlays from the directory, ordered lexicographically
+        then
+          let content = readDir dir; in
+          map (n: import (dir + ("/" + n)))
+            (builtins.filter (n: builtins.match ".*\.nix" n != null || pathExists (dir + ("/" + n + "/default.nix")))
+              (attrNames content))
+        # it's a file, so the result is the contents of the file itself
+        else import path;
     in
-      if dirPath != "" then
-        overlays dirPath
-      else if dirCheck dirHome then overlays dirHome
+      if pathOverlays != "" then
+        overlays pathOverlays
+      else if pathExists homeOverlaysFile then overlays homeOverlaysFile
```

Comment:
Thanks, good point. It should be agnostic for the value of `<nixpkgs-overlays>`, but not for the locations in the home directory.